core/src/main/codegen/templates/Parser.jj

```diff
@@ -6316,11 +6316,16 @@ SqlNode BuiltinFunctionCall() :
             if (flag == null) {
                 flag = SqlTrimFunction.Flag.BOTH.symbol(SqlParserPos.ZERO);
             }
+        }
+        [
+            <COMMA> {
+                trimChars = Expression(ExprContext.ACCEPT_SUB_QUERY);
+            }
+        ]
```

Comment:
This should not be allowed in the case that we've seen a FROM 'x'. I've pushed a test to show the problem